make/conf/module-loader-map.conf

```diff
@@ -91,3 +91,50 @@ PLATFORM_MODULES= \
 PLATFORM_MODULES_windows= \
     jdk.crypto.mscapi \
     #
+
+NATIVE_ACCESS_MODULES= \
+    java.base \
+    java.datatransfer \
+    java.desktop \
+    java.instrument \
+    java.logging \
+    java.management \
+    java.management.rmi \
+    java.naming \
+    java.net.http \
+    java.prefs \
+    java.rmi \
+    java.scripting \
+    java.se \
+    java.security.jgss \
+    java.security.sasl \
+    java.smartcardio \
+    java.sql \
+    java.sql.rowset \
+    java.transaction.xa \
+    java.xml \
+    java.xml.crypto \
+    jdk.accessibility \
+    jdk.charsets \
+    jdk.crypto.cryptoki \
+    jdk.dynalink \
+    jdk.httpserver \
+    jdk.incubator.vector \
+    jdk.internal.vm.ci \
+    jdk.jfr \
+    jdk.jsobject \
+    jdk.localedata \
+    jdk.management \
+    jdk.management.agent \
+    jdk.management.jfr \
+    jdk.naming.dns \
+    jdk.naming.rmi \
+    jdk.net \
+    jdk.nio.mapmode \
+    jdk.sctp \
+    jdk.security.auth \
+    jdk.security.jgss \
+    jdk.unsupported \
+    jdk.xml.dom \
+    jdk.zipfs \
```

Comment:
We should create a JBS issue to prune this.

Reply:
Many of these modules do not need native access in the current implementation.   Should this list be trimmed to list the ones that need native access in the current implementation?

Reply:
> Many of these modules do not need native access in the current implementation.

In addition this will eventually need jlink support. I view the change to ModuleBootstrap initialiser to use ModuleLoaderMap.nativeAccessModules() as very temporary. It may include many standard/JDK modules that aren't in the image. In addition we'll need some way to grant native access at link-time. The workaround for the latter right now is to configure default options.

Reply:
> Many of these modules do not need native access in the current implementation. Should this list be trimmed to list the ones that need native access in the current implementation?

Not sure if I know enough to do the pruning, so I was hoping that could be done separately (I'd file a bug as Alan suggests). But I can try to prune the list, if you prefer.

Reply:
Native access is needed for modules which calls restricted methods [1].   AFAICT, java.base, java.desktop and jdk.incubator.vector use java.lang.foreign but I don't know if they call restricted methods or not.

https://download.java.net/java/early_access/jdk23/docs/api/java.base/java/lang/foreign/package-summary.html#restricted

Reply:
> Native access is needed for modules which calls restricted methods [1].

In time, we'll need it for modules using JNI too. We can do this trimming in another PR to avoid Jan getting pulled into deeply.

Reply:
Doing it in another PR is fine with me.